src/Framework/MockObject/MockBuilder.php

```diff
@@ -178,6 +178,33 @@ public function setMethods(array $methods = null): self
         return $this;
     }
 
+    /**
+     * Specifies the subset of methods to mock, requiring each to exist in the class
+     *
+     * @param string[] $methods
+     */
+    public function setRealMethods(array $methods): self
+    {
+        if ($methods) {
+            $reflection = new \ReflectionClass($this->type);
+
+            foreach ($methods as $method) {
+                if (!$reflection->hasMethod($method)) {
+                    throw new RuntimeException(
+                        \sprintf(
+                            'Trying to set mock method "%s" which cannot be configured because it does not exist',
```

Comment:
Can we also have the class name in the exception message? This would make debugging problems easier.

Reply:
Good idea. I added it into the message and also changed up the message a little.